MaxText/layers/linears.py

```diff
@@ -382,17 +490,47 @@ def deepseek_scale_weights(self, weights):
     weights *= self.config.routed_scaling_factor
     return weights
 
-  def permute(self, inputs, gate_logits):
+  def deepseek_routing(self, gate_logits, pre_bias_logits):
+    """DeepSeek routing logit."""
```

Comment:
Some comments here?

e.g. 
This involves a two-stage selection: first selecting top groups of experts,
then selecting the top experts within those chosen groups. Selection uses
post-bias/post-score_func logits, but the returned weights are based on
pre-bias/pre-score_func logits.